### DISCUSSION TITLE
Logs report that aten_sink:get_failure_probabilities times out often

### DISCUSSION THREAD

**Discussion by sjzghw123** [rabbitmq/rabbitmq-server#11211] [Questions]
#Environmental information
RabbitMQ 3.9.5 on Erlang 24.1.6 Windows

#ERROR MESSAGE
** Generic server aten_detector terminating
** Last message in was poll
** When Server state == {state,#Ref<0.1631167254.3982491673.171500>,5000,0.99,
                               #{},#{}}
** Reason for termination ==
** {{timeout,{gen_server,call,[aten_sink,get_failure_probabilities]}},
    [{gen_server,call,2,[{file,"gen_server.erl"},{line,239}]},
     {aten_detector,handle_info,2,[{file,"src/aten_detector.erl"},{line,103}]},
     {gen_server,try_dispatch,4,[{file,"gen_server.erl"},{line,695}]},
     {gen_server,handle_msg,6,[{file,"gen_server.erl"},{line,771}]},
     {proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,226}]}]}

  crasher:
    initial call: aten_detector:init/1
    pid: <0.18404.627>
    registered_name: aten_detector
    exception exit: {timeout,
                        {gen_server,call,
                            [aten_sink,get_failure_probabilities]}}
      in function  gen_server:call/2 (gen_server.erl, line 239)
      in call from aten_detector:handle_info/2 (src/aten_detector.erl, line 103)
      in call from gen_server:try_dispatch/4 (gen_server.erl, line 695)
      in call from gen_server:handle_msg/6 (gen_server.erl, line 771)
    ancestors: [aten_sup,<0.174.0>]
    message_queue_len: 1
    messages: [poll]
    links: [<0.175.0>]
    dictionary: []
    trap_exit: false
    status: running
    heap_size: 4185
    stack_size: 29
    reductions: 52102
  neighbours:

supervisor: {local,aten_sup}
errorContext: child_terminated
reason: {timeout,{gen_server,call,[aten_sink,get_failure_probabilities]}}
offender: [{pid,<0.18404.627>},
           {id,aten_detector},
           {mfargs,{aten_detector,start_link,[]}},
           {restart_type,permanent},
           {significant,false},
           {shutdown,5000},
           {child_type,worker}]

**Comment by mkuratczyk** [comment]
RabbitMQ 3.9 has been out of support for quite some time.
https://www.rabbitmq.com/release-information

**Reply by sjzghw123** [reply]
I know it's already EOL, but would like to understand if this is a bug in the version

**Reply by michaelklishin** [reply]
@sjzghw123 so you do realize that the version has [long been out of support](https://www.rabbitmq.com/release-information) and you do realize that [you are not eligible for community support](https://github.com/rabbitmq/rabbitmq-server/blob/main/COMMUNITY_SUPPORT.md) but you figured'd ask anyway because the rules open source software maintainers set are meant to be ignored and broken.

Everything in OSS RabbitMQ is open source and developed in the open on GitHub. Everything, including dependencies. You have the same GitHub search as the core team. You could have copied the module name `aten_sink` or the function name `get_failure_probabilities` and [could have done a search](https://github.com/search?q=org%3Arabbitmq+get_failure_probabilities&type=issues) but you figured you'd ask anyway because those open source maintainers obviously owe you something.

Here, in a few clicks you arrive [at a very relevant discussion](https://github.com/rabbitmq/rabbitmq-server/discussions/5227). Your node is very likely out of CPU resources often enough for [RabbitMQ's failure detector](https://github.com/rabbitmq/aten) used by Raft-based features to time out. See [monitoring data](https://www.rabbitmq.com/docs/monitoring), increase the number of available cores (listed in `rabbitmq-diagnostics cluster_status` in modern versions) and [follow the docs](https://www.rabbitmq.com/docs/runtime#cpu-reduce-idle-usage) if you have a lot of queues, channels, connections that are mostly idle.
